Lib/test/test_gc.py

```diff
@@ -959,6 +964,69 @@ def getstats():
 
         gc.enable()
 
+    @unittest.skipIf(ContainerNoGC is None,
+                     'requires ContainerNoGC extension type')
+    def test_trash_weakref_clear(self):
```

Comment:
This test is great! Thanks for the time working on this. I find the comments very clear and I know is hard to control all these moving pieces. Great work!